src/matrix.jl

```diff
@@ -105,13 +107,14 @@ an operator of size `(N, N)` where `N = size(diag, 1)` is the leading length of
 `L` then is the elementwise-scaling operation on arrays of `length(u) = length(diag)`
 with leading length `size(u, 1) = N`.
 """
-function DiagonalOperator(diag::AbstractVector; update_func = DEFAULT_UPDATE_FUNC)
-    diag_update_func = if update_func == DEFAULT_UPDATE_FUNC
-        DEFAULT_UPDATE_FUNC
+function DiagonalOperator(diag::AbstractVector; update_func=nothing, accepted_kwargs=())
+    _update_func = preprocess_update_func(update_func, accepted_kwargs)
+    diag_update_func = if update_func_isconstant(_update_func)
+        _update_func
     else
-        (A, u, p, t) -> (update_func(A.diag, u, p, t); A)
+        (A, u, p, t; kwargs...) -> (_update_func(A.diag, u, p, t; kwargs...); A)
     end
-    MatrixOperator(Diagonal(diag); update_func=diag_update_func)
+    MatrixOperator(Diagonal(diag); update_func=diag_update_func, accepted_kwargs=nothing)
```

Comment:
`accepted_kwargs` is set to `()` at some places and to `nothing` at others. Can you pick one? either works.

Reply:
`nothing` actually means something very different: it's an unsafe signal "don't filter this function, it's already been filtered".

I think it might be too confusing for this to be `nothing`, since in SciML that usually means that a default will be chosen for it. Would you be okay with `SciMLOperators.NoFilter()`?

Reply:
Thanks for the clarification. `NoKwargFilter` should be good. :D